catalyst-gateway/bin/src/service/common/responses/mod.rs

```diff
@@ -134,12 +134,10 @@ impl<T> WithErrorResponses<T> {
     pub(crate) fn service_unavailable(err: &anyhow::Error, retry: RetryAfterOption) -> Self {
         let error = ServiceUnavailable::new(None);
         error!(id=%error.id(), error=?err, retry_after=?retry);
-        let retry = match retry {
-            RetryAfterOption::Default => Some(RetryAfterHeader::default()),
-            RetryAfterOption::None => None,
-            RetryAfterOption::Some(value) => Some(value),
-        };
-        WithErrorResponses::Error(ErrorResponses::ServiceUnavailable(Json(error), retry))
+        WithErrorResponses::Error(ErrorResponses::ServiceUnavailable(
+            Json(error),
+            retry.into(),
```

Comment:
revert.  Out of scope for ticket.
Backend will never set this.  Its controlled by the ingress.